matcher/match.go

```diff
@@ -85,25 +86,56 @@ func matchRectangular(pos position.Position, filter *repository.Filter) (bool, e
 	return xor(matched, filter.Reversed), nil
 }
 
-// matchRound has issue with matching over twelve meridian.
-// Return false when position and filter center are on different sides of it
 func matchRound(pos position.Position, filter *repository.Filter) (bool, error) {
 	var rfilter repository.RoundFilter
+	var limit float64 = 360
 	if err := json.Unmarshal(filter.Configuration, &rfilter); err != nil {
 		return false, err
 	}
-
-	matched := (pos.Latitude-rfilter.CenterLatitude)*(pos.Latitude-rfilter.CenterLatitude)+
+	preMatched := (pos.Latitude-rfilter.CenterLatitude)*(pos.Latitude-rfilter.CenterLatitude)+
 		(pos.Longitude-rfilter.CentreLongitude)*(pos.Longitude-rfilter.CentreLongitude) <=
 		(rfilter.Radius * rfilter.Radius)
-
-	return xor(matched, filter.Reversed), nil
+	if preMatched {
+		return xor(preMatched, filter.Reversed), nil
+	}
+	switch {
+	//round filter longitude and lalilude in critical position(near to +-180)
+	case (limit/2-math.Abs(float64(rfilter.CenterLatitude))-float64(rfilter.Radius)) <= 0 && (limit/2-math.Abs(float64(rfilter.CentreLongitude))-float64(rfilter.Radius)) <= 0:
```

Comment:
WDYT about using geometry?
https://www.khanacademy.org/math/geometry/hs-geo-analytic-geometry/hs-geo-dist-problems/v/point-relative-to-circle

Reply:
em...but i'm using geometry.
jsut we have cases whan point could be in pos -179 0 and filter in pos 179 0 with radius 10. and in our coordinates system the next numper after 180 is -179. in this case usual geometry doesn't work bcz (0-0)^2 +(179-(-179))^2 - 10^2 >0 
so this case help to calculate critical sections whan filter is near to 180 or -180 and calculate right result(true) by changing usual (x1-x2)^2 to (360- abs(x1)-abs(x2))^2 for this ordinate.
usual calculation i use in 4th case of switch